mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -47,29 +47,34 @@ def find_mlir_build_dir() -> str:
     Finds mlir build dir searching either WORKSPACE dir
     or home dir
     """
-    miopen_opt_path = None
+    miopen_gen_path = None
     candidate_paths = [
         # if the script is run from build dir
-        Path('./bin/miopen-opt'),
+        Path('./bin/miopen-gen'),
         # if the script is run from source
-        Path(__file__).parent.parent.parent.parent / 'build' / 'bin' / 'miopen-opt'
+        Path(__file__).parent.parent.parent.parent / 'build' / 'bin' / 'miopen-gen'
     ]
     for candidate_path in candidate_paths:
         if candidate_path.exists():
-            miopen_opt_path = candidate_path
-    if not miopen_opt_path:
+            miopen_gen_path = candidate_path
+    
+    if not miopen_gen_path:
         try:
             # Prioritize the search in the current repo first.
             search_root = str(subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode().strip())
         except subprocess.CalledProcessError:
             # Else look in the home or WORKSPACE directory
             search_root = os.environ.get('WORKSPACE', str(Path.home()))
             assert search_root, "Cant find WORKSPACE env arg or home directory"
-        miopen_opt_path = glob.glob(search_root + '/**/bin/miopen-opt', recursive=True)
-        assert len(miopen_opt_path) == 1, "Multiple paths found to contain */bin/miopen-opt"
-        miopen_opt_path = miopen_opt_path[0]
+        
+        miopen_gen_path = glob.glob(search_root + '/**/bin/miopen-gen', recursive=True)
+        if len(miopen_gen_path) == 0:
+            # MLIR miopen_gen not available
+            return None
+        assert len(miopen_gen_path) == 1, "Multiple paths found to contain */bin/miopen-gen"
```

Comment:
To fix nightlies, instead of the assert (my bad!) lets return None -- so that the consumer can complain if mlir build is expected rather than prematurely assert here.

Reply:
It returns None when miopen_gen_path is not found. The assert is testing if we got multiple paths containing miopen-gen.

Reply:
Ack